workspace/build_target_test.go

```diff
@@ -0,0 +1,114 @@
+package workspace
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/yourbase/yb/runtime"
+)
+
+func TestBuildTarget_EnvironmentVariables(t *testing.T) {
+	type fields struct {
+		Name        string
+		Tools       []string
+		Commands    []string
+		Environment []string
+	}
+	type args struct {
+		data runtime.RuntimeEnvironmentData
+	}
+	// Sets up dotenv
+	cwd, err := filepath.Abs(".")
+	if err != nil {
+		t.Fatalf("Unable to find current work dir: %v", err)
+	}
+	testPath := filepath.Join(cwd, "TESTING")
+	err = os.Mkdir(testPath, 0755)
+	if err != nil {
+		t.Fatalf("Unable to make test dir: %v", err)
+	}
+	dotEnvFilePath := filepath.Join(testPath, ".env")
+	err = os.Chdir(testPath)
```

Comment:
Rather than calling `os.Chdir`, is there a way we can change the system-under-test to permit passing in a file path? IIUC the non-test callers of the function can just always pass in `".env"` because that will use the file in the current directory.

Reply:
Good call, yeah!